python/pyspark/sql/tests/test_python_streaming_datasource.py

```diff
@@ -150,6 +178,20 @@ def check_batch(df, batch_id):
         q.awaitTermination
         self.assertIsNone(q.exception(), "No exception has to be propagated.")
 
+    def test_simple_stream_reader(self):
+        self.spark.dataSource.register(self._get_simple_data_source())
+        df = self.spark.readStream.format("SimpleDataSource").load()
+
+        def check_batch(df, batch_id):
+            assertDataFrameEqual(df, [Row(batch_id * 2), Row(batch_id * 2 + 1)])
+
+        q = df.writeStream.foreachBatch(check_batch).start()
+        while len(q.recentProgress) < 10:
+            time.sleep(0.2)
+        q.stop()
+        q.awaitTermination
+        self.assertIsNone(q.exception(), "No exception has to be propagated.")
+
     def test_stream_writer(self):
         input_dir = tempfile.TemporaryDirectory(prefix="test_data_stream_write_input")
```

Comment:
let's add cleanup

```python
        input_dir.cleanup()
        output_dir.cleanup()
        checkpoint_dir.cleanup()
```

those into `finally`. Feel free to do it in a separate PR.

Reply:
Fixed.